lib/async-iterator.js

```diff
@@ -213,6 +221,16 @@ class MapIterator extends AsyncIterator {
   }
 }
 
+class ReverseIterator extends AsyncIterator {
+  constructor(base) {
+    const newBase = [];
```

Comment:
This should be done on the first `next` invocation. `base` is always `AsyncInterotor`.